app/controllers/brewpubs_controller.rb

```diff
@@ -1,5 +1,55 @@
-class BrewpubsController < ApplicationController
+class BrewpubsController <ApplicationController
+  before_action :authenticate_user!, except: [:index, :show]
+    before_action :authorize_user, except: [:index, :show, :new, :create]
+
   def index
-    @brewpubs = Brewpub.all
+    @brewpubs= Brewpub.all
+  end
+
+  def new
+    @brewpub = Brewpub.new
+  end
+
+  def create
+    @brewpub = Brewpub.create(brewpub_params)
+    @brewpub.user = current_user
+    if @brewpub.save
+      redirect_to root_path, notice: "Brewpub added successfully"
+    else
+      flash[:alert] = @brewpub.errors.full_messages.join(',')
+      render :new
+    end
+  end
+
+  def show
+    @brewpub = Brewpub.find(params[:id])
+    # @reviews = @brewpub.reviews
+  end
+
+  def edit
+    @brewpub = Brewpub.find(params[:id])
+  end
+
+  def update
+    @brewpub = Brewpub.find(params[:id])
+    # This doesn't save yet
```

Comment:
Be careful leaving comments in your code. You can commit features that aren't finished and put them in a separate branch

Reply:
makes sense